dsc/workflows/base/base_workflow.py

```diff
@@ -0,0 +1,199 @@
+
+        PLANNED CODE:
+
+        sqs_client = SQSClient()
+        for item_submission in self.item_submissions_iter():
+            item_submission.upload_dspace_metadata(
+                self.s3_bucket, item_submission.metadata_keyname
+            )
+            sqs_client.send_submission_message(
+                item_submission.item_identifier,
+                self.workflow_name,
+                self.output_queue,
+                self.submission_system,
+                self.collection_handle,
+                item_submission.metadata_uri,
+                item_submission.bitstream_uris,
+            )
+        """
+
+    @final
+    def item_submissions_iter(self) -> Iterator[ItemSubmission]:
+        """Generate a batch of item submissions for the DSpace Submission Service.
+
+        MUST NOT be overridden by workflow subclasses.
+        """
+        for item_metadata in self.batch_metadata_iter():
+            item_identifier = self.get_item_identifier(item_metadata)
+            logger.info(f"Processing submission for '{item_identifier}'")
+            metadata_keyname = f"{self.s3_prefix}/{item_identifier}_metadata.json"
+            dspace_metadata = self.create_dspace_metadata(item_metadata)
+            if self.validate_dspace_metadata(dspace_metadata):
+                item_submission = ItemSubmission(
+                    dspace_metadata=dspace_metadata,
+                    bitstream_uris=self.get_bitstream_uris(item_identifier),
+                    metadata_keyname=metadata_keyname,
+                )
+                yield item_submission
+
+    @abstractmethod
+    def batch_metadata_iter(self) -> Iterator[dict[str, Any]]:
+        """Iterate through batch metadata to yield item metadata.
+
+        MUST be overridden by workflow subclasses.
+        """
+
+    @abstractmethod
+    def get_item_identifier(self, item_metadata: Any) -> str:  # noqa: ANN401
+        """Get identifier for an item submission according to the workflow subclass.
+
+        MUST be overridden by workflow subclasses.
+
+        Args:
+            item_metadata: The item metadata from which the item identifier is extracted.
+        """
+
+    @final
+    def create_dspace_metadata(self, item_metadata: dict[str, Any]) -> dict[str, Any]:
+        """Create DSpace metadata from the item's source metadata.
+
+        A metadata mapping is a dict with the format seen below:
+
+        {
+        "dc.contributor": {
+            "source_field_name": "contributor",
+            "language": None,
+            "delimiter": "|",
+        }
+
+        MUST NOT be overridden by workflow subclasses.
+
+        Args:
+            item_metadata: Item metadata from which the DSpace metadata will be derived.
+        """
+        metadata_entries = []
+        for field_name, field_mapping in self.metadata_mapping.items():
+            if field_name not in ["item_identifier", "source_system_identifier"]:
```

Comment:
Why would `item_identifier` or `source_system_identifier` be present in the mapping in the first place?

It seems as if the mapping would be constructed specifically for this application, and therefore could be omitted.

Reply:
`item_identifier` is provided by stakeholders in the CSV but may require additional parsing in the `get_item_identifier` method beyond retrieving it from the CSV (e.g. could be the full filename when we want to omit the extension). 

`source_system_identifier` is for creating reports used to update related metadata in other repositories (e.g. a digital object URI ArchivesSpace is the big use case) with the newly published DSpace handle. So it needs to be there for linking but isn't a part of the DSpace metadata that's generated. My expectation is this will be done in the `finalize` command. But second thought, it might be better to omit `source_system_identifier` until I'm ready to fully implement that functionality in the `ArchivesSpace` workflow, thoughts?

Reply:
> But second thought, it might be better to omit source_system_identifier until I'm ready to fully implement that functionality in the ArchivesSpace workflow, thoughts?

Might not be a bad idea?  We've been trying a similar approach -- and it's hard -- in the parquet work, to keep the early passes really tight and focused.  I think it's okay to stub something when it's almost gauranteed to show up later, and perhaps the contours of seeing it helps understand the current code to review.  But yeah, maybe it'll be easier to reason about, name, and organize when things settle into place a bit more.  Would make sense to me to wait.

Reply:
Removed!